crates/cxx-qt-lib/src/core/qtlogging.rs

```diff
@@ -0,0 +1,129 @@
+// SPDX-FileCopyrightText: 2025 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// SPDX-FileContributor: Joshua Goins <joshua.goins@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+use cxx::{type_id, ExternType};
+use std::ffi::c_char;
+use std::ffi::CStr;
+use std::marker::PhantomData;
+
+#[cxx::bridge]
+mod ffi {
+    /// The level the message is sent to the message handler at.
+    #[repr(i32)]
+    enum QtMsgType {
+        /// A debug message.
+        QtDebugMsg = 0,
+        /// An info message.
+        QtInfoMsg = 4,
+        /// A warning message.
+        QtWarningMsg = 1,
+        /// A fatal message.
+        QtFatalMsg = 3,
+        /// A critical message.
+        QtCriticalMsg = 2,
```

Comment:
question while waiting: do we rename enum values to something more sensible? in this case the names are weird to avoid collisions, since in qtlogging.h these were just an `enum`. hence why there' prefixed with `Qt` and suffixed with `Msg`.